codec-http/src/main/java/io/netty/handler/codec/http/QueryStringDecoder.java

```diff
@@ -224,6 +280,10 @@ private void decodeParams(String s) {
                     name = decodeComponent(s.substring(pos, i), charset);
                 }
                 pos = i + 1;
+            } else if (withCommaDecoding && c == ',' && name != null) {
```

Comment:
Nice hedge case.

We have to invent something here as there's no spec mentioning how one should interpret a comma in a query. The RFC3986 only specifies that comma is a valid char in there (but only gives clue on how to use it for path parameters, not query). And HTML's `application/x-www-form-urlencoded` requires comma to be percent encoded.

I'd tend to stick close to `application/x-www-form-urlencoded` spec and only interpret comma as a delimiter inside a value. So here we'd have `((",", "f"))`.